src/xbot/common/controls/sensors/MockXboxController.java

```diff
@@ -12,10 +15,74 @@ public MockXboxController(int port) {
     XYPair leftJoystickAxis;
     XYPair rightJoystickAxis;
     double leftTriggerAxis;
+    double x = 0;
+    double y = 0;
     double rightTriggerAxis;
+    private boolean xRightInverted = false;
+    private boolean yRightInverted = false;
+    private boolean xLeftInverted = false;
+    private boolean yLeftInverted = false;
+    
+    Map<Integer, Boolean> buttons = new HashMap<Integer, Boolean>();
+    Map<Integer, Double> rawAxis = new HashMap<Integer, Double>();
     
     boolean[] buttonValues = new boolean[10];
     
+    public void setRightX(double x) {
+        double value = x * (xRightInverted ? -1 : 1);
+        this.x = value;
+        setRawAxis(0, value);
+    }
+
+    public void setLeftX(double x) {
+        double value = x * (xLeftInverted ? -1 : 1);
+        this.x = value;
+        setRawAxis(0, value);
+    }
+    
+    public void setRightY(double y) {
+        double value = y * (yRightInverted ? -1 : 1);
+        this.y = value;
+        setRawAxis(1, value);
+    }
+
+    public void setLeftY(double y) {
+        double value = y * (yLeftInverted ? -1 : 1);
+        this.y = value;
+        setRawAxis(1, value);
+    }
+    
+    public void setRawAxis(int which, double value) {
+        rawAxis.put(which, value);
+    }
+
+    public void pressButton(int button) {
+        buttons.put(button, true);
+    }
+
+    public void releaseButton(int button) {
+        buttons.put(button, false);
+    }
+
+    public XYPair getRightVector() {
+        return new XYPair(this.x * (xRightInverted ? -1 : 1), this.y
+                * (yRightInverted ? -1 : 1));
+    }
+
+    public XYPair getLeftVector() {
+        return new XYPair(this.x * (xLeftInverted ? -1 : 1), this.y
```

Comment:
These should use the internal vectors instead of these variables. As-is, both sticks return the same values.